kernel/page_table_entry/src/lib.rs

```diff
@@ -67,7 +68,17 @@ impl PageTableEntry {
 
     /// Returns this `PageTableEntry`'s flags.
     pub fn flags(&self) -> PteFlagsArch {
-        PteFlagsArch::from_bits_truncate(self.0)
+        #[cfg(target_arch = "x86_64")]
+        {
+            PteFlagsArch::from_bits_truncate(self.0)
+        }
+
+        #[cfg(target_arch = "aarch64")]
+        {
+            let mut arch_flags = PteFlagsArch::from_bits_truncate(self.0);
+            arch_flags.remove(PteFlagsArch::PAGE_DESCRIPTOR);
```

Comment:
hm, I thought we wanted this bit to be set, since it indicates a page descriptor instead of a block descriptor. (I thought a block descriptor was similar to huge pages on x86)

Is this correct, or am I misunderstanding this flag's meaning? Here's what we wrote in `pte_flags_aarch64`:
```rust
        /// * If set, this represents a page descriptor.
        /// * If not set, this represents a block descriptor.
        const PAGE_DESCRIPTOR    = 1 << 1;
````

Reply:
It looks like I introduced a regression in that branch while I've been reverting the workspace merging commit... I'll double check the revert to see if this is my only mistake.